pkg/tls/tls.go

```diff
@@ -93,28 +105,41 @@ func GetServerConfig(serverLog *zerolog.Logger, client kclient.Client) (*tls.Con
 		}
 		return config, nil
 	}
+
 	serverLog.Info().Msg(MTLSEnabledMsg)
-	CACertsData, err := GetCertificatesFromSecret(client, caSecretName, caSecretNamespace)
+	cacertProvided, err := isCertificateProvided(caSecretName, caSecretNamespace)
 	if err != nil {
+		serverLog.Error().Err(err).Msg("error in the provided CA certificate")
 		return nil, err
 	}
-	CACertPool := x509.NewCertPool()
-	for key, element := range CACertsData {
-		// skip non cerificate keys like crt.key if exists in the secret
-		if !strings.HasSuffix(key, TLSCertKeySuffix) {
-			continue
+	var CACertPool *x509.CertPool
+	if cacertProvided {
+		CACertsData, err := GetCertificatesFromSecret(client, caSecretName, caSecretNamespace)
+		if err != nil {
+			return nil, err
 		}
-		if !CACertPool.AppendCertsFromPEM(element) {
-			serverLog.Error().Err(err).Msg(err.Error())
-			return nil, errors.New("error in GetServerConfig in AppendCertsFromPEM trying to lead key:" + key)
+		CACertPool, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+		for key, element := range CACertsData {
+			// skip private key
+			if key == corev1.TLSPrivateKeyKey {
+				continue
+			}
+			if !CACertPool.AppendCertsFromPEM(element) {
+				serverLog.Error().Err(err).Msg(err.Error())
+				return nil, errors.New("error in GetServerConfig in AppendCertsFromPEM trying to lead key:" + key)
+			}
 		}
 	}
 
 	config = &tls.Config{
 		Certificates: []tls.Certificate{loadedCertServer},
-		ClientAuth:   tls.RequireAndVerifyClientCert,
-		ClientCAs:    CACertPool,
-		MinVersion:   tls.VersionTLS13,
+		// configure mutual TLS
+		ClientAuth: tls.RequireAndVerifyClientCert,
+		ClientCAs:  CACertPool,
```

Comment:
Can you check, please, that when you add empty CACertPool, you don't overwrite the default setting ?